KMB/make_KMB_info.py

```diff
@@ -0,0 +1,270 @@
+        :return: str
+        """
+        return helpers.format_filename(item.get_description(), 'KMB', item.ID)
+
+    # @todo:
+    # * Add a collaboration + COH template in the source field - T164569
+    # * Add the source link to the source field - T164569
+    def make_info_template(self, item):
+        """
+        Create the description template for a single KMB entry.
+
+        :param item: the metadata for the media file in question
+        :return: str
+        """
+        template_name = 'Kulturmiljöbild-image'
+        template_data = OrderedDict()
+        template_data['short title'] = item.namn
+        template_data['original description'] = item.beskrivning
+        template_data['wiki description'] = item.get_wiki_description()
+        template_data['photographer'] = self.get_photographer(item)
+        template_data['depicted place'] = item.get_depicted_place()
+        template_data['date'] = item.date
+        template_data['permission'] = item.license
+        template_data['ID'] = item.ID
+        template_data['bildbeteckning'] = item.bildbeteckning
+        template_data['notes'] = ''
+        txt = helpers.output_block_template(template_name, template_data, 0)
+
+        # append object location if appropriate
+        if item.latitude and item.longitude:
+            txt += '\n{{Object location dec|%s|%s}}' % (
+                item.latitude, item.longitude)
+
+        return txt
+
+    def get_original_filename(self, item):
+        """
+        Return the url from which the image is fetched.
+
+        :param item: the KMBItem
+        :return: str
+        """
+        return item.source
+
+    # @todo: load mapping list instead of substing template - T164567
+    def get_photographer(self, item):
+        """
+        Return a correctly formated photographer value
+
+        :param item: the KMBItem
+        :return: str
+        """
+        template = 'User:Lokal_Profil/nycklar/creators'
+        return '{{safesubst:%s|%s|t}}' % (template, item.byline)
+
+    # @todo:
+    # * use fmis + bbr for (cached) commonscat searches on wikidata. And if
+    #   found then skip various other. - T164572
+    # * move some of this to KMBItem and set cats at the same time as the
+    #   text is processed.
+    # * Implement taggs from mapping - T164567
+    # * Add parish/municipality categorisation when needed - T164576
+    def generate_content_cats(self, item):
+        """
+        Extract any mapped keyword categories or depicted categories.
+
+        :param item: the KMBItem to analyse
+        :return: list of categories (without "Category:" prefix)
+        """
+        cats = []
+
+        # depicted
+        if item.fmis:
+            cats.append('Archaeological monuments in %s' % item.landskap)
+            cats.append('Archaeological monuments in %s County' % item.lan)
+        if item.bbr:
+            cats.append('Protected buildings in Sweden')
```

Comment:
I think I had originally skipped this since the WLM bot nicely categorises these images later anyway (this happens for all bbr but only for the fmis ones which are included in WLM).

To be honest I had forgotten about this assumption so there are some tasks related to doing better categorisation (for both bbr and fmis). The following are the main ones [T164719](https://phabricator.wikimedia.org/T164719), [T164717](https://phabricator.wikimedia.org/T164717), [T164572](https://phabricator.wikimedia.org/T164572). Since partial categorisation messes with the WLM bot categorisation I'll have to have a think to see if this is actually worth doing.